docs/vue.md

```diff
@@ -20,118 +20,314 @@ To get started, load either the production or development version of Vue in your
+  {{ count }}
+  <button v-on:click="count += 1">+</button>
+</div>
+```
 
-  <button v-on:click="hello">Say Hello</button>
+<output id="counter">
+  <button v-on:click="count -= 1">-</button>
+  {{ count }}
+  <button v-on:click="count += 1">+</button>
+</output>
+
+## Components
+
+Use `vueComponents` to register Vue components using the component name as the key with an object containing Vue options as the value. Components data is not shared and changes will not persist as users navigate the site.
+
+```js
+window.$docsify = {
+  vueComponents: {
+    'button-counter': {
+      template: `
+        <button @click="count += 1">
+          You clicked me {{ count }} times
+        </button>
+      `,
+      data() {
+        return {
+          count: 0,
+        };
+      },
+    },
+  },
+};
+```
 
-  <button v-on:click="counter -= 1">-</button>
-  {{ counter }}
-  <button v-on:click="counter += 1">+</button>
-</div>
+```markdown
+<button-counter></button-counter>
+<button-counter></button-counter>
+```
 
-<!-- prettier-ignore-end -->
+<output data-lang="output">
+  <button-counter></button-counter>
+  <button-counter></button-counter>
+</output>
+
+## Markdown script
+
+Vue content can mounted using a `<script>` tag in your markdown pages.
 
-!> Only the first `<script>` tag in a markdown file is executed. If you are working with multiple Vue components, all `Vue` instances must be created within this tag.
+!> Only the first `<script>` tag in a markdown file is executed. If you wish to mount multiple Vue instances using a script tag, all instances must be mounted within the first script tag in your markdown.
 
+```html
+<!-- Vue 2.x  -->
 <script>
   new Vue({
-    el: "#example-1",
-    data: function() {
-      return {
-        counter: 0,
-        message: "Hello, World!"
-      };
-    },
-    methods: {
-      hello: function() {
-        alert(this.message);
-      }
-    }
+    el: '#example',
+    // Options...
   });
 </script>
+```
+
+```html
+<!-- Vue 3.x  -->
+<script>
+  Vue.createApp({
+    // Options...
+  }).mount('#example');
+</script>
+```
+
+## Technical Notes
+
+- Docsify processes Vue content in the following order:
+  1. Markdown `<script>`
+  1. `vueMountOptions`
+  1. `vueGlobalOptions`
+- Docsify will not mount an existing Vue instance or an element that contains an existing Vue instance.
+- Docsify will automatically destroy/unmount all Vue instances it creates before new page content is loaded.
+- When processing `vueGlobalOptions`, docsify parses each root element of the main content area and mounts the element if Vue content is detected. Docsify does not parse each individual node within the main content area.
+- When processing `vueGlobalOptions`, docsify will only detect the full `v-` attribute syntax (e.g `v-bind:href` or `v-on:click`). For performance reasons, detection of Vue's [shorthand](https://vuejs.org/v2/guide/syntax.html#Shorthands) attribute syntax (e.g. `:href`or `@click`) is not supported.
```

Comment:
What performance reasons are these? I always use shorthand for everything in Vue and have been fine in all my cases so far (performance wise).

Reply:
There are performance implications for docsify, not for Vue.

The issue is that I'm parsing content for Vue syntax to determine which elements to mount and pass `vueGlobalOptions` to. This is done using regular expressions:

https://github.com/docsifyjs/docsify/blob/a5a431f0c156119a6fd6fbea53b67e7eb745d550/src/core/render/index.js#L129-L131

The full syntax is the most efficient and reliable to parse (above). The shorthand syntax is less efficient and either less reliable or impossible to parse:

#### `v-bind` Shorthand

```html
<!-- full syntax -->
<a v-bind:href="url"> ... </a>

<!-- shorthand -->
<a :href="url"> ... </a>

<!-- shorthand with dynamic argument (2.6.0+) -->
<a :[key]="url"> ... </a>
```

#### `v-on` Shorthand

```html
<!-- full syntax -->
<a v-on:click="doSomething"> ... </a>

<!-- shorthand -->
<a @click="doSomething"> ... </a>

<!-- shorthand with dynamic argument (2.6.0+) -->
<a @[event]="doSomething"> ... </a>
```

I'd prefer to release something strict but efficient and reliable, then wait for feedback from users to determine if we need to support shorthand syntax as well. If we have to, we can, but doesn't feel like a requirement at this point for me.

Reply:
Added support for Vue shorthand directive syntax in https://github.com/docsifyjs/docsify/pull/1409/commits/35ad5c559b43f6d70fae054b1812edef9d2b7e4d.